### PR TITLE
machine: Add append, owner, and permission parameters to SSHConnection.write()

### PR DESCRIPTION
**Pull request by martinpitt**
These are a common operations in tests, let's make them declarative.

---

I tested this in a local rewrite of https://github.com/cockpit-project/cockpit/pull/14264 .

### CODE REVIEW

File: machine/machine_core/ssh_connection.py [hunk 1]
```diff
@@ -437,17 +437,21 @@ def download_dir(self, source, dest, relative_dir="."):
         except:
             self.message("Error while downloading directory '{0}'".format(source))
 
-    def write(self, dest, content):
+    def write(self, dest, content, append=False, mode=None):
```
Comment (allisonkarlitskaya) [inline machine/machine_core/ssh_connection.py:440]: `chmod` might be a better name than `mode`.  it's less accurate, but it's much clearer.  `mode` is too overloaded.  `perm` is another possibility.  `mode` is also fine.  your choice :)

`owner=` (or similarly, `chown=`) string e.g. `admin:admin` or so, might be a nice addition.

File: machine/machine_core/ssh_connection.py [hunk 2]
```diff
@@ -437,17 +437,21 @@ def download_dir(self, source, dest, relative_dir="."):
...
         """Write a file into the test machine
 
         Arguments:
             content: Raw data to write to file
             dest: The file name in the machine to write to
+            append: If True, append to existing file instead of replacing it
+            mode: Optional file access mode as shell string (e.g. "0600")
```
Comment (allisonkarlitskaya) [inline machine/machine_core/ssh_connection.py:447]: `append` and `chmod` together is strange to imagine...

File: machine/machine_core/ssh_connection.py [hunk 3]
```diff
@@ -437,17 +437,21 @@ def download_dir(self, source, dest, relative_dir="."):
...
         """
         assert dest
         assert self.ssh_address
 
-        cmd = "cat > '%s'" % dest
+        cmd = "cat %s '%s'" % (append and '>>' or '>', dest)
+        if mode:
+            cmd += ' && chmod %s %s' % (mode, dest)
```
Comment (allisonkarlitskaya) [inline machine/machine_core/ssh_connection.py:454]: Please use an equivalent amount of quoting for both appearances of `dest`

File: machine/machine_core/ssh_connection.py (hunk 1 above)
Comment (martinpitt) [reply machine/machine_core/ssh_connection.py]: Right, perm= makes sense, and I added owner= as well. We don't use it anywhere right now, but at least it's there when we want to.

File: machine/machine_core/ssh_connection.py (hunk 2 above)
Comment (martinpitt) [reply machine/machine_core/ssh_connection.py]: True that, same for append+owner, but *technically* it works.

File: machine/machine_core/ssh_connection.py (hunk 3 above)
Comment (martinpitt) [reply machine/machine_core/ssh_connection.py]: Well spotted, done.

**Pull request merged**